src/host.c

```diff
@@ -202,6 +206,11 @@ void HostInitConfig(char quiet)
             SC_ATOMIC_SET(host_config.memcap, host_memcap);
         }
     }
+    if (GlobalMemcapReached(SC_ATOMIC_GET(host_config.memcap))) {
```

Comment:
could you create a unit func for this? Right now we see this same code block many times. E.g.:
```c
    GlobalMemcapSomename(SC_ATOMIC_GET(host_config.memcap), "host.memcap");
```
Error handling and 'exit' could then be handled in the new util func.